ouroboros-consensus/test-storage/Test/Ouroboros/Storage/FS/StateMachine.hs

```diff
@@ -191,7 +191,7 @@ newtype Resp fp h = Resp { getResp :: Either FsError (Success fp h) }
 
 -- | The 'Eq' instance for 'Resp' uses 'sameFsError'
 instance (Eq fp, Eq h) => Eq (Resp fp h) where
-  Resp (Left  e) == Resp (Left  e') = sameFsError e e'
```

Comment:
It would be nice to be a little bit more precise here (as precise as we can be without making `MockFS` OS dependent I guess). But this can be done in a separate PR.

Reply:
Concretely: let's replace `FsIllegalOperation`, `FsResourceInappropriateType` and `FsInvalidArgument` by a more general `FsInvalidOperation`, and then introduce a Posix and Windows specific `ioToFsError` function. Ideally we'd _not_ bundle `FsInsufficientPermissions` in there though, so if you do try this and find some problems with that, let's think about the specific problems there.